### ISSUE TITLE
HuggingFace model generation only works if tokenizer is invoked before usage in Guidance

### ISSUE THREAD

**Issue by StevenSong** [guidance-ai/guidance#1322]
**The bug**
I have a fine-tuned GPT2-derived model based on `openai-community/gpt2`. The trained model is on huggingface under `labadvcomp/gdc-cohort-llm-gpt2-1m-dev`. I'm using it with `guidance` to generate structured JSONs with a Pydantic model. I preload the model and tokenizer before passing both to `guidance.models.Transformers` (would like to keep this pattern instead of passing the model path to preserve some custom logic for model device mapping).

However, loading the tokenizer fresh and passing it to `Transformers` results in an exception:
```
Exception: Attempted to run a transformers model past its maximum context window size of 1024!
```

I've found that if I instead load the model from the base model, it is able to terminate generation without exceeding the max context length. In the model training, I did not retrain or modify the tokenizer except for re-saving it with the trained model weights. There's no difference in the tokenizer output (same token ids, attention masks, etc)

The interesting part is that it seems like I can prevent the error when using the tokenizer saved with my model simply by invoking the tokenizer once on some dummy filler text, before passing it to `Transformers`. I've implemented this as my work-around in my application since it's a very cheap operation, but it seems odd that simply invoking the tokenizer would lead to a change in behavior. I'm very unfamiliar with the implementation of huggingface's fast tokenizers though so maybe this is obvious behavior I'm misunderstanding.

**To Reproduce**
```python
from guidance.models import Transformers
from transformers import GPT2TokenizerFast, GPT2LMHeadModel
from guidance import json as gen_json

from typing import Annotated, Literal
from pydantic import BaseModel, StringConstraints

class InnerStrContent(BaseModel):
    field: Annotated[str, StringConstraints(max_length=128)]
    value: list[Annotated[str, StringConstraints(min_length=1, max_length=128)]]


class Inner(BaseModel):
    op: Literal["in"]
    content: InnerStrContent


class CohortSchema(BaseModel):
    op: Literal["and"]
    content: list[Inner]


tok = GPT2TokenizerFast.from_pretrained("labadvcomp/gdc-cohort-llm-gpt2-1m-dev")
# tok = GPT2TokenizerFast.from_pretrained("openai-community/gpt2") # using this works without pre-invocation
model = GPT2LMHeadModel.from_pretrained("labadvcomp/gdc-cohort-llm-gpt2-1m-dev")
model = model.eval()

# tok("foobar") # pre-invoking the tokenizer fixes the error seen when using the saved tokenizer

lm = Transformers(
    model=model,
    tokenizer=tok,
)
lm += "bam files"
lm += gen_json(
    name="cohort", schema=CohortSchema, temperature=0, max_tokens=1024,
)
```

expected output is:
```
`{"op": "and", "content": [{"op": "in", "content": {"field": "cases.files.data_format", "value": ["bam"]}}]}`
```

**System info (please complete the following information):**
 - OS (e.g. Ubuntu, Windows 11, Mac OS, etc.): Ubuntu 20.04.6
 - Guidance Version (`guidance.__version__`): `0.2.4`

**Comment by hudson-ai** [comment]
Thanks for the issue @StevenSong !

This is quite odd... 

No concrete theories yet, just some observations. For a Fast tokenizer like this one, we get a json-serialized dump of the underlying rust tokenizer, including added tokens from `tokenizer_config.json`, like so:

```python
stringified = tok.backend_tokenizer.to_str()
```

I'm noticing that the returned json is *different* depending on whether or not the tokenizer has been pre-invoked. In particular:

Pre:
```json
{
  "truncation": {
    "direction": "Right",
    "max_length": 1024,
    "strategy": "LongestFirst",
    "stride": 0
  },
  "padding": {
    "strategy": {
      "Fixed": 1024
    },
    "direction": "Right",
    "pad_to_multiple_of": null,
    "pad_id": 50256,
    "pad_type_id": 0,
    "pad_token": "<|endoftext|>"
  },
  ...
}
```

Post:
```json
{
  "truncation": null,
  "padding": null,
  ...
}
```

That padding seems to be the issue... It seems that invoking `tok.encode` actually *modifies* this. Will look into this further when I have the opportunity.
